test/convert/to_text_test.cc

```diff
@@ -0,0 +1,1020 @@
+        text::Event{text::EventDesc{
+                        nullopt, At{loc2,
+                                    text::EventType{
+                                        EventAttribute::Exception,
+                                        text::FunctionTypeUse{
+                                            At{loc3, text::Var{Index{0}}},
+                                            {},
+                                        },
+                                    }}},
+                    {}}},
+     At{loc1, binary::Event{At{loc2, binary::EventType{
+                                         EventAttribute::Exception,
+                                         At{loc3, Index{0}},
+                                     }}}});
+}
+
+TEST(ConvertToTextTest, Module) {
+  // Additional locations only needed for Module. :-)
+  const SpanU8 loc9 = "I"_su8;
+  const SpanU8 loc10 = "J"_su8;
+  const SpanU8 loc11 = "K"_su8;
+  const SpanU8 loc12 = "L"_su8;
+  const SpanU8 loc13 = "M"_su8;
+  const SpanU8 loc14 = "N"_su8;
+  const SpanU8 loc15 = "O"_su8;
+  const SpanU8 loc16 = "P"_su8;
+  const SpanU8 loc17 = "Q"_su8;
+  const SpanU8 loc18 = "R"_su8;
+  const SpanU8 loc19 = "S"_su8;
+  const SpanU8 loc20 = "T"_su8;
+  const SpanU8 loc21 = "U"_su8;
+  const SpanU8 loc22 = "V"_su8;
+  const SpanU8 loc23 = "W"_su8;
+  const SpanU8 loc24 = "X"_su8;
+  const SpanU8 loc25 = "Y"_su8;
+  const SpanU8 loc26 = "Z"_su8;
+  const SpanU8 loc27 = "AA"_su8;
+  const SpanU8 loc28 = "BB"_su8;
+
+  auto binary_table_type =
+      At{"T0"_su8, binary::TableType{At{"T1"_su8, Limits{At{"T2"_su8, u32{0}}}},
+                                     At{"T3"_su8, bt::RT_Funcref}}};
+
+  auto text_table_type =
+      At{"T0"_su8, text::TableType{At{"T1"_su8, Limits{At{"T2"_su8, u32{0}}}},
+                                   At{"T3"_su8, TRT_Funcref}}};
+
+  auto memory_type =
+      At{"M0"_su8, MemoryType{At{"M1"_su8, Limits{At{"M2"_su8, u32{0}}}}}};
+
+  auto binary_global_type =
+      At{"G0"_su8, binary::GlobalType{At{"G1"_su8, bt::VT_I32},
+                                      At{"G2"_su8, Mutability::Const}}};
+
+  auto text_global_type = At{
+      "G0"_su8,
+      text::GlobalType{At{"G1"_su8, TVT_I32}, At{"G2"_su8, Mutability::Const}}};
+
+  auto external_kind = At{"EK"_su8, ExternalKind::Function};
+
+  // These will be shared between global, data, and element segments. This
+  // would never actually happen, but it simplifies the test below.
+  auto binary_constant_expression =
+      At{"CE0"_su8,
+         binary::ConstantExpression{
+             At{"CE1"_su8, binary::Instruction{At{"CE2"_su8, Opcode::I32Const},
+                                               At{"CE3"_su8, s32{0}}}}}};
+
+  auto text_constant_expression =
+      At{"CE0"_su8,
+         text::ConstantExpression{
+             At{"CE1"_su8, text::Instruction{At{"CE2"_su8, Opcode::I32Const},
+                                             At{"CE3"_su8, s32{0}}}}}};
+
+  OK(At{loc1,
+        text::Module{
```

Comment:
Wow, this deeply nested initializers are like a DSL of sorts.   <meme>Thats gross, I Love it</meme>